core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -284,47 +286,73 @@ class LogCleaner(initialConfig: CleanerConfig,
 
     /**
      * The main loop for the cleaner thread
+     * Clean a log if there is a dirty log available, otherwise sleep for a bit
      */
     override def doWork() {
-      cleanOrSleep()
+      val cleaned = cleanFilthiestLog()
+      if (!cleaned)
+        pause(config.backOffMs, TimeUnit.MILLISECONDS)
     }
 
     /**
-     * Clean a log if there is a dirty log available, otherwise sleep for a bit
-     */
-    private def cleanOrSleep() {
-      val cleaned = cleanerManager.grabFilthiestCompactedLog(time) match {
-        case None =>
-          false
-        case Some(cleanable) =>
-          // there's a log, clean it
-          var endOffset = cleanable.firstDirtyOffset
-          try {
-            val (nextDirtyOffset, cleanerStats) = cleaner.clean(cleanable)
-            recordStats(cleaner.id, cleanable.log.name, cleanable.firstDirtyOffset, endOffset, cleanerStats)
-            endOffset = nextDirtyOffset
-          } catch {
-            case _: LogCleaningAbortedException => // task can be aborted, let it go.
-            case _: KafkaStorageException => // partition is already offline. let it go.
-            case e: IOException =>
-              val msg = s"Failed to clean up log for ${cleanable.topicPartition} in dir ${cleanable.log.dir.getParent} due to IOException"
-              logDirFailureChannel.maybeAddOfflineLogDir(cleanable.log.dir.getParent, msg, e)
-          } finally {
-            cleanerManager.doneCleaning(cleanable.topicPartition, cleanable.log.dir.getParentFile, endOffset)
+      * Cleans a log if there is a dirty log available
+      * @return whether a log was cleaned
+      */
+    private def cleanFilthiestLog(): Boolean = {
```

Comment:
I'd have loved to replace `filthiest` with something better but I can't think of much. `cleanDirtyLog`, may be?

Reply:
`filthiest` and `filthy` is used in multiple places to denote a dirty compacted log:
https://github.com/apache/kafka/blob/9f7267dd2fedde86bf15aabdbc5256e5fc617184/core/src/main/scala/kafka/log/LogCleanerManager.scala#L122
https://github.com/apache/kafka/blob/9f7267dd2fedde86bf15aabdbc5256e5fc617184/core/src/main/scala/kafka/log/LogCleanerManager.scala#L144
and all over the test.

I think if we'd change the naming we'd need to change some of those occurrences as well. May I ask why you prefer dirty over filthiest? I see both as synonyms